### ISSUE TITLE
Integratig a function with singularities using the quad routine

### ISSUE THREAD

**Issue by TMueller83** [scipy/scipy#7555]
I am using the `quad` function from `scipy.integrate v0.19.1` to integrate functions with a square-root like singularity at each end of the integration interval such as for example
```
In [1]: quad(lambda x: 1/sqrt(1-x**2), -1, 1)
```

(I use the `sqrt` function from `numpy v1.12.0`) which immediately yields the correct result pi:
```
Out[1]: (3.141592653589591, 6.200897573194197e-10)
```
According to the documentation of the `quad` function the keyword `points` should be used to indicate the locations of singularities or discontinuities of the integrand, but if I indicate the points `[1, -1]` where the above integrand is singluar I get a Warning and `nan` as result:
```
In [2]: quad(lambda x: 1/sqrt(1-x**2), -1, 1, points=[-1, 1])
    
RuntimeWarning: divide by zero encountered in double_scalars
IntegrationWarning: Extremely bad integrand behavior occurs at some points of the integration interval.

Out[2]: (nan, nan)
```
Why does `quad` produce these problems if the singularities of the integrand are specified and just runs fine if these points are not indicated?

**EDIT:** I figured out, that the correct way to handle the singularities in the above integral is to use a proper weighting function. For the above integral I can use
```
quad(lambda x: 1, -1, 1, weight='alg', wvar=(-1/2, -1/2))
```
and everything is is fine. The above problem with the keyword `points` is probably due to the fact that a different QUADPACK routine is used if `points` are specified.

**Comment by rgommers** [comment]
Thanks @TMueller83. That does seem a little confusing. It would probably help to add some examples to the docstring that deal with functions with singularities.

**Comment by jcalbert** [comment]
Looked a bit more into this.  It seems the problem is that if `points` contains `a`, `b`, or any duplicates, then QUADPACK's QAGPE routine will try to integrate a zero-width subinterval.  So in your case, it's trying to integrate the subintervals `[-1,-1]`,`[-1,1]` and `[1,1]`.  Doing so forces a function evaluation at `-1` even though Gauss-Kronrod usually avoids endpoints.

Using the weighted function is probably preferable, but I think a small fix is in order anyway.

**Comment by ilayn** [comment]
Fixed in #7975

**Comment by TMueller83** [comment]
I can confirm that this is fixed in scipy v1.1.0. Thanks to the developers!

But for anyone interested I would like to point out, that the use of weighting functions for this kind of integrals should still be preferred, since (at least in my simple example)
1) The result is more precise:
```
quad(lambda x: 1/sqrt(1-x**2), -1, 1, points=[-1, 1])
(3.141592653589591, 6.200897573194197e-10)

quad(lambda x: 1, -1, 1, weight='alg', wvar=(-1/2, -1/2))
(3.1415926535897927, 9.860180640534107e-14)
```

2) Execution time is much faster:
```
%timeit quad(lambda x: 1/sqrt(1-x**2), -1, 1, points=[-1, 1])
1.09 ms ± 6.96 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)

%timeit quad(lambda x: 1, -1, 1, weight='alg', wvar=(-1/2, -1/2))
16.8 µs ± 13.5 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)
```
